app-games/src/main/java/com/aptoide/android/aptoidegames/AptoideAsyncImage.kt

```diff
@@ -73,6 +93,96 @@ fun AptoideAsyncImage(
   )
 }
 
+@Composable
+fun AptoideAsyncImageWithFullscreen(
+  modifier: Modifier = Modifier,
+  data: Any?,
+  placeholder: Boolean = true,
+  contentDescription: String?,
+  transformations: Transformation? = null,
+  colorFilter: ColorFilter? = null,
+  screenshots: List<String?> = emptyList(),
+  selectedIndex: Int = 1,
+) {
+  val placeholderColor = Palette.Grey
+  var expanded by remember { mutableStateOf(false) }
+  AsyncImage(
+    model = buildModel(data, transformations),
+    placeholder = if (placeholder) remember { ColorPainter(placeholderColor) } else null,
+    error = remember { ColorPainter(placeholderColor) },
+    contentDescription = contentDescription,
+    contentScale = ContentScale.Crop,
+    colorFilter = colorFilter,
+    modifier = modifier.clickable { expanded = true },
+  )
+
+  if (expanded) {
+    FullscreenImageViewer(
+      screenshots = screenshots,
+      initialPage = selectedIndex,
+      onDismiss = { expanded = false }
+    )
+  }
+}
+
+@Composable
+fun FullscreenImageViewer(
+  screenshots: List<String?>,
+  initialPage: Int,
+  onDismiss: () -> Unit
+) {
+  AlertDialog(
+    onDismissRequest = onDismiss,
+    buttons = {},
+    properties = DialogProperties(usePlatformDefaultWidth = false),
+    modifier = Modifier.fillMaxSize(),
+    text = {
+      Box(
+        modifier = Modifier.fillMaxSize()
+      ) {
+        Image(
+          imageVector = getLeftArrow(Palette.Primary, Palette.Black),
+          contentDescription = stringResource(id = R.string.button_back_title),
+          contentScale = ContentScale.Crop,
+          modifier = Modifier
+            .clickable(onClick = onDismiss)
+            .size(32.dp)
+        )
+        val pagerState = rememberPagerState(
+          initialPage = initialPage,
+          pageCount = { screenshots.size }
+        )
+        val zoomState = rememberZoomState(
+          maxScale = 5f
+        )
+
+        LaunchedEffect(pagerState.currentPage) {
+          zoomState.reset()
+        }
+
+        HorizontalPager(
+          state = pagerState,
+          modifier = Modifier.fillMaxSize()
+        ) { page ->
+          Box(
```

Comment:
Each image should have a limited zooming height, so that when zooming it does not overlap the back button